server/channels/app/platform/web_conn.go

```diff
@@ -789,6 +800,13 @@ func (wc *WebConn) ShouldSendEvent(msg *model.WebSocketEvent) bool {
 			wc.lastAllChannelMembersTime = 0
 		}
 
+		// Execute channel hook
+		if msg.GetBroadcast().ChannelHook != nil && msg.GetBroadcast().ChannelHook(wc.UserId, msg) {
+			// If hook returns true, that means message has been modified. We need
+			// to wipe off the pre-computed JSON
+			msg.RemovePrecomputedJSON()
```

Comment:
Why don't we leave this burden to the hook definition? On that side we have access to the message and the necessary context to make the decision right?

Reply:
That's why I am returning the true/false. It didn't look to me something that the hook should be aware about. The app developer shouldn't be aware about precomputed JSON. That's why I chose to do it in the web_conn layer.